core/src/Cuda/Kokkos_Cuda_abort.hpp

```diff
@@ -28,35 +28,23 @@ extern "C" {
 /*  Cuda runtime function, declared in <crt/device_runtime.h>
  *  Requires capability 2.x or better.
  */
-extern __device__ void __assertfail(const void *message, const void *file,
-                                    unsigned int line, const void *function,
-                                    size_t charsize);
+__device__ [[noreturn]] void __assertfail(const void *message, const void *file,
+                                          unsigned int line,
+                                          const void *function,
+                                          size_t charsize);
 }
 
 namespace Kokkos {
 namespace Impl {
 
 // required to workaround failures in random number generator unit tests with
 // pre-volta architectures
-#if defined(KOKKOS_ENABLE_DEBUG_BOUNDS_CHECK)
-__device__ inline void cuda_abort(const char *const message) {
-#else
-[[noreturn]] __device__ inline void cuda_abort(const char *const message) {
-#endif
+__device__ __noinline__ [[noreturn]] static void cuda_abort(
```

Comment:
```suggestion
[[noreturn]] __device__ __noinline__ static void cuda_abort(
```

Reply:
Oh neat, it seemed like that worked, thank you :)